nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-jsf/src/test/java/org/nuxeo/ecm/imaging/recompute/lang/test/PerformingImageActionTest.java

```diff
@@ -0,0 +1,149 @@
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.core.api.security.impl.ACLImpl;
+import org.nuxeo.ecm.core.api.security.impl.ACPImpl;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.platform.actions.ActionContext;
+import org.nuxeo.ecm.platform.actions.ejb.ActionManager;
+import org.nuxeo.ecm.platform.actions.jsf.JSFActionContext;
+import org.nuxeo.ecm.platform.ui.web.jsf.MockFacesContext;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class })
+@Deploy("org.nuxeo.ecm.actions")
+@Deploy("org.nuxeo.ecm.webapp.core:OSGI-INF/actions-contrib.xml")
+@Deploy("org.nuxeo.ecm.platform.tag:OSGI-INF/faceted-tag-service-core-types.xml")
+@Deploy("org.nuxeo.ecm.platform.picture.core:OSGI-INF/picture-schemas-contrib.xml")
+@Deploy("org.nuxeo.ecm.platform.picture.jsf")
+public class PerformingImageActionTest {
+
+    public static final String PICTURE_NAME = "anyPicture";
+
+    public static final String PICTURE_DOC_TYPE = "Picture";
+
+    public static final String USER_WITH_READ_PERMISSION = "robert";
+
+    public static final String USER_WITH_READ_WRITE_PERMISSION = "james";
+
+    public static final String USER_WITH_EVERYTHING_PERMISSION = "anotherAdmin";
+
+    public static final String ROTATE_PICTURE_FILTER_NAME = "canRotatePicture";
+
+    @Inject
+    protected ActionManager actionService;
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    protected MockFacesContext facesContext;
+
+    @Before
+    public void initRepository() {
+        facesContext = new MockFacesContext();
+        facesContext.setCurrent();
+
+        DocumentModel document = coreSession.createDocumentModel("/", PICTURE_NAME, PICTURE_DOC_TYPE);
+        coreSession.createDocument(document);
+
+        ACE[] aces = { getACE(USER_WITH_READ_PERMISSION, SecurityConstants.READ),
+                getACE(USER_WITH_READ_WRITE_PERMISSION, SecurityConstants.READ_WRITE),
+                getACE(USER_WITH_EVERYTHING_PERMISSION, SecurityConstants.EVERYTHING) };
+
+        ACLImpl acl = new ACLImpl();
+        acl.setACEs(aces);
+
+        ACPImpl acp = new ACPImpl();
+        acp.addACL(acl);
+        document.setACP(acp, Boolean.FALSE);
+
+        coreSession.save();
+    }
+
+    @Test
+    public void shouldBeRotatableByTheCreator() {
+        boolean canRotatePicture = actionService.checkFilter(ROTATE_PICTURE_FILTER_NAME, getActionContext(coreSession));
```

Comment:
I'm not sure we need that boolean variable, it would be enough to do:
```suggestion
        assertTrue(actionService.checkFilter(ROTATE_PICTURE_FILTER_NAME, getActionContext(coreSession)));
```